### PR TITLE
Fix issue with `concat` on maps returning keys instead of key/values

### LINKED ISSUE #871: `concat` on maps issue
Hi,

`concat` on maps wrongly concatenates the map keys rather that the key value pairs.

To reproduce.
1. Open up the REPL and concatenate a couple of one item maps, it returns back the list of keys rather than the key value pairs
``` clojure
basilisp.user=> (concat {:a 1} {:b 2})
(:a :b)
```

Contrast this with Clojure
``` clojure
user> (concat {:a 1} {:b 2})
([:a 1] [:b 2])
```

PR to follow.

### PR DESCRIPTION
**Pull request by ikappaki**
Hi,

could you please consider patch to have `concat` on maps return k/v pairs instead of keys. It fixes #871.

The issue as I understand it that the underlying map library iterates over keys rather than k/v pairs, and thus it needs to be converted first to a sequence (which correctly converts map into a seq of pairs).

Thanks

### CODE REVIEW

File: src/basilisp/lang/runtime.py [hunk 1]
```diff
@@ -1073,7 +1073,8 @@ def cons(o, seq) -> ISeq:
 
 def concat(*seqs: Any) -> ISeq:
     """Concatenate the sequences given by seqs into a single ISeq."""
-    return lseq.sequence(itertools.chain.from_iterable(filter(None, seqs)))
+    s = [to_seq(seq) if isinstance(seq, IPersistentMap) else seq for seq in seqs if seq]
+    return lseq.sequence(itertools.chain.from_iterable(s))
```
Comment (chrisrink10) [inline src/basilisp/lang/runtime.py:1077]:
```suggestion
    return lseq.sequence(itertools.chain.from_iterable(filter(None, map(to_seq, seqs))))
```

I believe this should be correct in all cases.

File: tests/basilisp/runtime_test.py
Comment (chrisrink10) [inline tests/basilisp/runtime_test.py:1]: Can you add test a test case where a string is one of the elements being concatenated? Likewise with a `None` (`nil`) value. Both should be valid and appear to already work but aren't tested.

**Review by chrisrink10 — COMMENTED** [review]
Thanks for the patch. I had a suggestion for a more concise fix that I think should also fix the reported issue.

File: src/basilisp/lang/runtime.py (hunk 1 above)
Comment (ikappaki) [reply src/basilisp/lang/runtime.py]: Updated

File: tests/basilisp/runtime_test.py
Comment (ikappaki) [reply tests/basilisp/runtime_test.py:1]: Added an additional test case, which includes strings and Nones, I hope this is what you had in mind, thanks

Comment (chrisrink10) [reply tests/basilisp/runtime_test.py:1]: Yes, that looks great. Thanks!

**Review by chrisrink10 — APPROVED** [review]
